meson.build

```diff
@@ -0,0 +1,91 @@
+# SPDX-License-Identifier: LGPL-2.1 OR BSD-2-Clause
+project('libbpf', 'c',
+        version : '0.0.3',
+        license : 'LGPL-2.1 OR BSD-2-Clause',
+        default_options : [
+                'prefix=/usr',
+        ],
+        meson_version : '>= 0.46',
+        )
+
+patchlevel = '0'
+extraversion = '3'
+
+libbpf_source_dir = './'
+
+libbpf_sources = files(run_command('find',
+        [
+        '@0@/src'.format(libbpf_source_dir),
+        '-type',
+        'f',
+        '-name',
+        '*.[h|c]']).stdout().split())
+
+libbpf_headers = files(
+        join_paths(libbpf_source_dir, 'src/bpf.h'),
+        join_paths(libbpf_source_dir, 'src/btf.h'),
+        join_paths(libbpf_source_dir, 'src/libbpf.h'))
+
+feature_rellocarray = run_command(join_paths(libbpf_source_dir, 'scripts/check-reallocarray.sh'))
+
+libbpf_c_args = ['-g',
+                '-O2',
+                '-Werror',
+                '-Wall',
+]
+
+if feature_rellocarray.stdout().strip() != ''
+        libbpf_c_args += '-DCOMPAT_NEED_REALLOCARRAY'
+endif
+
+# bpf_includes are required to include bpf.h, btf.h, libbpf.h
+bpf_includes = include_directories(
+        join_paths(libbpf_source_dir, 'src'))
+
+libbpf_includes = include_directories(
+        join_paths(libbpf_source_dir, 'include'),
+        join_paths(libbpf_source_dir, 'include/uapi'))
+
+libelf_dep = dependency('libelf')
+deps = [libelf_dep]
+
+libbpf_static = static_library(
+        'bpf',
+        libbpf_sources,
+        c_args : libbpf_c_args,
+        dependencies : deps,
+        include_directories : libbpf_includes,
+        install : true)
+
+libbpf_static_dep = declare_dependency(link_with : libbpf_static)
+
+libbpf_map_source_path = join_paths(libbpf_source_dir, 'src/libbpf.map')
+libbpf_map_abs_path = join_paths(meson.current_source_dir(), libbpf_map_source_path)
+
+libbpf_c_args += ['-fPIC', '-fvisibility=hidden']
+
+libbpf_link_args = ['-Wl,--version-script=@0@'.format(libbpf_map_abs_path)]
+
+libbpf_shared = shared_library(
+        'bpf',
+        libbpf_sources,
+        c_args : libbpf_c_args,
+        dependencies : deps,
+        include_directories : libbpf_includes,
+        install : true,
+        link_args : libbpf_link_args,
+        link_depends : libbpf_map_source_path,
+        soversion : patchlevel,
+        version : extraversion)
```

Comment:
@wat-ze-hex symlinks are still created differently:
```
diff -u <(cd src && stat -c %N libbpf.so* | sort) <(cd build && stat -c %N libbpf.so* | sort)
--- /dev/fd/63  2019-04-15 10:20:53.029672412 -0700
+++ /dev/fd/62  2019-04-15 10:20:53.030672409 -0700
@@ -1,3 +1,3 @@
-‘libbpf.so.0.0.3’
-‘libbpf.so.0’ -> ‘libbpf.so.0.0.3’
+‘libbpf.so.0’ -> ‘libbpf.so.3’
+‘libbpf.so.3’
 ‘libbpf.so’ -> ‘libbpf.so.0’
```

See [corresponding code](https://github.com/libbpf/libbpf/blob/8bbb21c227d64efe3ad96cfd427984e51f6ee549/src/Makefile#L73-L82) in `src/Makefile`.

I assume only `extraversion` is used where the whole version should be.

Reply:
My bad. Fixed:
```
$ diff -u <(cd src && stat -c %N libbpf.so* | sort) <(cd build && stat -c %N libbpf.so* | sort)
$
```